modules/openapi-generator/src/main/resources/dart2/api.mustache

```diff
@@ -185,19 +185,18 @@ class {{{classname}}} {
     // At the time of writing this, `dart:convert` will throw an "Unexpected end of input"
     // FormatException when trying to decode an empty string.
     if (response.body != null && response.statusCode != HttpStatus.noContent) {
-      {{#isArray}}
-      return (apiClient.deserialize(_decodeBodyBytes(response), '{{{returnType}}}') as List)
-        .cast<{{{returnBaseType}}}>()
-        .{{#uniqueItems}}toSet(){{/uniqueItems}}{{^uniqueItems}}toList(growable: false){{/uniqueItems}};
-      {{/isArray}}
-      {{^isArray}}
-        {{#isMap}}
-      return {{{returnType}}}.from(apiClient.deserialize(_decodeBodyBytes(response), '{{{returnType}}}'));
-        {{/isMap}}
-        {{^isMap}}
-      return apiClient.deserialize(_decodeBodyBytes(response), '{{{returnType}}}') as {{{returnType}}};
-        {{/isMap}}
-      {{/isArray}}
+    {{#isArray}}
```

Comment:
You can maybe check `isContainer` first to make the else branch clearer.
And `List/Set.of` can be used instead of `from`.

Reply:
I think the best way to do the template branches here is using `isContainer`, `isSimpleType` and `isPrimitiveType` - that should give you all 3 cases with needing to negate.